packages/file-asset-apis/src/file-api/file-reader-api.ts

```diff
@@ -0,0 +1,124 @@
+import { TSError } from '@terascope/job-components';
+import fse from 'fs-extra';
+import { FileSlicer } from './file-slicer';
+import { SlicedFileResults } from '../interfaces';
+import { ChunkedFileReader, segmentFile, canReadFile } from '../lib';
+
+export class FileReader extends ChunkedFileReader {
+    client = fse;
+
+    /**
+     * low level api that fetches the unprocessed contents of the file, please use the "read" method
+     * for correct file and data parsing
+     * @example
+     * const slice = { offset: 0, length: 1000, path: 'some/file.txt', total: 1000 };
+     * const results = await fileReader.fetch(slice);
+     * results === 'the unprocessed contents of the file here'
+    */
+    protected async fetch(slice: SlicedFileResults): Promise<string> {
+        const { path, length, offset } = slice;
+        const fd = await fse.open(path, 'r');
+
+        try {
+            const buf = Buffer.alloc(2 * this.config.size);
+            const { bytesRead } = await fse.read(fd, buf, 0, length, offset);
+            return this.decompress(buf.slice(0, bytesRead));
+        } finally {
+            fse.close(fd);
+        }
+    }
+    /**
+     * Determines if a file name or file path can be processed, it will return false
+     * if the name of path includes a "."
+     *
+     * @example
+     * fileReader.canReadFile('file.txt')  => true
+     * fileReader.canReadFile('some/path/file.txt')  => true
+     * fileReader.canReadFile('some/.private_path/file.txt')  => false
+    */
+    canReadFile(fileName: string): boolean {
+        return canReadFile(fileName);
+    }
+
+    /**
+     * Determines if a directory can be processed, will throw if it is a symlink
+     * or if the directory is empty
+     *
+     * @example
+     *
+     * fileReader.validatePath('some/emptyDir') => Error
+     * fileReader.canReadFile('some/symlinkedDir')  => Error
+     * fileReader.canReadFile('some/dir') => void
+    */
+    validatePath(path: string): void {
+        try {
+            const dirStats = fse.lstatSync(path);
+
+            if (dirStats.isSymbolicLink()) {
+                throw new TSError({ reason: `Directory '${path}' cannot be a symlink!` });
```

Comment:
That is an invalid use of `TSError`, also don't add `!` to error messages.

```suggestion
                throw new Error(`Directory '${path}' cannot be a symlink`);
```